main.go

```diff
@@ -400,30 +362,25 @@ func (a *Analyzer) analyze() {
 	a.createSyntheticTests()
 
 	failedTestNamesAcrossAllJobRuns := getFailedTestNamesFromJobRuns(a.RawData.JobRuns)
-	updateGlobalBugCache(failedTestNamesAcrossAllJobRuns)
+	a.BugCache.UpdateForFailedTests(failedTestNamesAcrossAllJobRuns.List()...)
 
 	// for every test that failed in some job run, look up the bug(s) associated w/ the test
 	// and attribute the number of times the test failed+flaked to that bug(s)
-	for t := range failedTestNamesAcrossAllJobRuns {
-		if util.IgnoreTestRegex.MatchString(t) {
+	for testName := range failedTestNamesAcrossAllJobRuns {
+		if util.IgnoreTestRegex.MatchString(testName) {
```

Comment:
can this be just `util.IsTestIgnored()` ?

Reply:
> can this be just `util.IsTestIgnored()` ?

this shouldn't leak out.  I tightened it up in the interim